app/views/admin/events/_all_with_comments.csv.haml

```diff
@@ -0,0 +1,23 @@
+- headers = ['Event ID',
+             'Title',
+             'Abstract',
+             'Start time',
+             'Submitter',
+             'Speaker',
+             'Event Type',
+             'Track',
+             'Difficulty Level',
+             'Room',
+             'State',
+             'Comments']
+= CSV.generate_line ["Events with Comments"]
+= CSV.generate_line headers
+- @events.each do |event|
+  - all_comments = ''
+  - event.root_comments.each do |comment|
+    - all_comments << "#{comment.created_at.strftime("%Y-%m-%d")} #{comment.created_at.strftime("%I:%M%p")}  #{comment.user.name}: #{comment.body}\n"
+  = CSV.generate_line([event.id, event.title, event.abstract, (event.time.present? ? "#{event.time.strftime("%Y-%m-%d")} #{event.time.strftime("%I:%M%p")} " : 'nil'),
+                       event.submitter.name, event.speaker_names, event.event_type.title,
+                       (event.track.present? ? event.track.name : 'nil'),
+                       (event.difficulty_level.present? ? event.difficulty_level.title : 'nil'),
+                       (event.room.present? ? event.room.name : 'nil'), event.state, all_comments]).html_safe
```

Comment:
Why do you put `html_safe` only here and not in the other options as well?

It would be very useful if you could try your code in your instance, to see if everything looks OK. CSV export does work properly in the other options. Please try it with realistic data, ie a formatted abstract